crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,150 @@
+//! Run a prepass before the main pass to generate depth and/or normals texture, also called a g-buffer.
+//! These textures are useful for various screen-space effects and reducing overdraw in the main pass.
+//!
+//! To enable the prepass, you need to add a prepass component to a [`crate::prelude::Camera3d`].
+//!
+//! [`DepthPrepass`]
+//! [`NormalPrepass`]
+//!
+//! Both textures are available on the `PrepassTextures` component attached to each [`crate::prelude::Camera3d`] with a prepass component.
+//! The textures are automatically added to the default mesh view bindings. You can also get the raw textures
+//! by querying the [`ViewPrepassTextures`] component on any camera with a prepass component.
+//!
+//! When using the default mesh view bindings you should be able to use `prepass_depth()`
+//! and `prepass_normal()` to load the related textures. These functions are defined in `bevy_pbr::utils`.
+//! See the `shader_prepass` example that shows how to use it.
+//!
+//! The prepass runs for each `Material`, you can control if the prepass should run by setting the `prepass_enabled`
+//! flag on the `MaterialPlugin`.
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_reflect::Reflect;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d, TextureFormat},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+pub const DEPTH_PREPASS_FORMAT: TextureFormat = TextureFormat::Depth32Float;
+pub const NORMAL_PREPASS_FORMAT: TextureFormat = TextureFormat::Rgb10a2Unorm;
+
+/// If added to a [`crate::prelude::Camera3d`] then depth values will be copied to a separate texture available to the main pass.
+/// The main pass already uses a depth texture by default which helps reduce overdraw, but this will help reduce it even more.
+///
+/// Make sure to enable the prepass on your `Material` for this to do anything.
```

Comment:
should we remove this (and the one below) since they are on by default?

Reply:
My intuition was that it's still worth pointing it out since it's technically possible to disable it and forget that it is disabled. I don't really mind either way.

Reply:
It would need to be rephrased a bit though, so I ended up just removing them.